src/liquid/components/ld-tabs/ld-tablist/ld-tablist.tsx

```diff
@@ -112,32 +115,90 @@ export class LdTablist {
       ?.focus()
   }
 
-  private onKeydown(ev) {
+  private onClick = (ev) => {
+    if (ev.pointerType === 'mouse') {
+      this.focusVisible = false
+    }
+  }
+
+  private onFocusout = (ev) => {
+    if (
+      !ev.relatedTarget ||
+      ev.relatedTarget.closest('ld-tablist') !== this.el
+    ) {
+      this.focusVisible = true
+      this.selectedIsFocused = true
+    }
+  }
+
+  private onKeydown = (ev) => {
     switch (ev.key) {
       case 'ArrowLeft':
         ev.preventDefault()
+        this.focusVisible = true
         this.focusTab(ev.target, 'left')
         return
       case 'ArrowRight': {
         ev.preventDefault()
+        this.focusVisible = true
         this.focusTab(ev.target, 'right')
         return
       }
       case 'ArrowDown': {
         ev.preventDefault()
+        this.focusVisible = true
         this.setFocusOnSelectedTabpanel()
         return
       }
     }
   }
 
+  private handleResize = () => {
+    if (this.mode === 'floating') {
+      this.updateSelectedTabIndicator()
+    } else {
+      this.updateScrollable()
+      this.updateScrollButtons()
+    }
+  }
+
+  private updateSelectedTab = () => {
+    this.selectedTab = Array.from(this.el.querySelectorAll('ld-tab')).find(
+      (tab) => tab.selected
+    )
+  }
+
+  @Listen('ldtabselect')
+  handleTabSelect(ev) {
+    this.selectedIsFocused = true
+    this.selectedTab = ev.target
+  }
+
+  @Watch('selectedTab')
+  private updateSelectedTabIndicator() {
+    if (!this.selectedTabIndicatorRef) return
```

Comment:
If this case is possible, it shouldn't be typed as

```ts
private selectedTabIndicatorRef!: HTMLElement
```

but without the `!`. Otherwise this condition is not needed.